tests/commands/run_test.py

```diff
@@ -211,6 +211,37 @@ def test_run(
     )
 
 
+def test_run_output_logfile(
+        cap_out,
+        tempdir_factory,
+        mock_out_store_directory,
+):
+
+    expected_output = (b'This is STDOUT output',
+                       b'This is STDERR output')
+
+    def raise_error_on_none(*args):
+        if len(args) and args[0] is None:
+            raise TypeError()
+        return mock.DEFAULT
+
+    with mock.patch('pre_commit.output.open', mock.mock_open()) as mocked_open:
```

Comment:
I'd rather see a real integration test which the file is actually written -- especially since the code as written doesn't properly write the file -- There's fixtures in testing/fixtures for setting up a git repository (and fixtures in testing/conftest.py as well)